### ISSUE TITLE
ID token verification is failing when id_token is signed with ES256

### ISSUE THREAD

**Issue by deshah04** [oauth2-proxy/oauth2-proxy#1626]
## Expected Behavior
ID token verification should pass if the token is signed with ES256 (bug)

This works fine when the prebuilt docker image [quay.io/oauth2-proxy/oauth2-proxy](https://quay.io/oauth2-proxy/oauth2-proxy) is used. It fails with the latest code changes.

## Current Behavior
ID token verification is failing with the following error:
[2022/04/22 11:34:24] [oauthproxy.go:768] Error redeeming code during OAuth2 callback: could not verify id_token: failed to verify token: oidc: id token signed with unsupported algorithm, expected ["RS256"] got "ES256"

## Possible Solution
Investigated this a bit, seems like it might be related to changes in OIDC discovery code (https://github.com/oauth2-proxy/oauth2-proxy/pull/1559)

Looked into the code via debugger,  From the Verifier code: (https://github.com/coreos/go-oidc/blob/v3/oidc/verify.go), it is taking the default signing algorithm as RS256, since SupportedSigningAlgs is empty.

```
	supportedSigAlgs := v.config.SupportedSigningAlgs
	if len(supportedSigAlgs) == 0 {
		supportedSigAlgs = []string{RS256}
	}

	if !contains(supportedSigAlgs, sig.Header.Algorithm) {
		return nil, fmt.Errorf("oidc: id token signed with unsupported algorithm, expected %q got %q", supportedSigAlgs, sig.Header.Algorithm)
	}
```

One possible fix is to pass SupportedSigAlgs (received from the wellknown endpoint during OIDC discovery) to Verifier (v.config) - similar to passing ClientID)

## Steps to Reproduce (for bugs)

1) Confirmed that provider wellknown endpoint returns:
"id_token_signing_alg_values_supported": [
    "ES256"
  ],

2) Confirmed that provider token endpoint returns ID_token signed with ES256. JOSE header contains the correct information, and kid matches with the keys retrieved from JWKS endpoint.

3) When trying to access the API, oauth2_proxy reaches the Provider using Auth Code PKCE flow, retrieves access/id_token. 
However, it fails to verify the id_token

## Context

Trying to use the latest code, since PKCE is needed for our use-case.
Provider configuration cannot be changed to sign id_token with RS256, hence this is a blocking issue.

- Version used: latest code
This issue is not observed when using the docker image quay.io/oauth2-proxy/oauth2-proxy

**Comment by JoelSpeed** [comment]
I believe this is what #1578 was trying to resolve. We need to fix this before we cut the new release

**Comment by JoelSpeed** [comment]
This should now be fixed by #1662
